pkg/manager/leaderelection.go

```diff
@@ -57,11 +57,11 @@ func (k *KubeMacPoolManager) UpdateLeaderLabel() error {
 				return errors.Wrap(err, "failed to get kubemacpool manager pod")
 			}
 
-			_, exist := pod.Labels[names.LEADER_LABEL]
+			_, exist := pod.GetLabels()[names.LEADER_LABEL]
 			if pod.Name == k.podName && !exist {
 				logger.V(1).Info("add the label to the elected leader", "Pod Name", pod.Name)
```

Comment:
can you remove `V(1)` here ? it does not happend very often and it helps debug d/s